pkg/controllers/cloud/networkpolicy_pendingitem.go

```diff
@@ -245,20 +245,6 @@ func (p *pendingGroup) RunOrDeletePendingItem(id string, context interface{}) (r
 func (p *pendingGroup) ClearPendingState() {}
 
 func (s *securityGroupImpl) runPendingItemImpl(c cloudSecurityGroup, memberOnly bool, r *NetworkPolicyReconciler) bool {
-	if s.retryOp == nil {
-		// retry operation successful. Reconcile current group with
-		// plug-ing
-		if s.state != securityGroupStateGarbageCollectState {
```

Comment:
why is this removed? can you explain the scenario?

Reply:
Because its causing potential extra cloud calls and I don't see it serves any purpose at all. If a retry succeeded, the operation will send cloudResponse and the state machine continues, no reason to reconcile again here. 

In network policy unit test create fail case, this code is causing race conditions. On create fail, the operation is retried. After retry success, the state machine continues to update rules, and this reconcile code is also triggered, which also calls update rules.